docs/source/topics/configfile.rst

```diff
@@ -40,11 +40,24 @@ a stage specific configuration value is needed, the ``stages`` mapping
 is checked first.  If no value is found then the top level keys will
 be checked.
 
-
 * ``api_gateway_stage`` - The name of the API gateway stage.  This
   will also be the URL prefix for your API
   (``https://endpoint/prefix/your-api``).
 
+* ``api_gateway_endpoint_type`` - The endpoint configuration of the
+  deployed API Gateway which determines how the API will be accessed,
+  can be EDGE, REGIONAL, PRIVATE. Note this value can only be set as a
+  top level key and defaults to EDGE. For more information see
+  https://amzn.to/2LofApt
+
+* ``api_gateway_endpoint_vpce`` - When configuring a Private API a VPC
+  Endpoint must be specified to configure a default resource policy on
+  the API if an explicit policy is not specified.
+
+* ``api_gateway_policy`` - The IAM resource policy for the REST API.
```

Comment:
For this variable, I think it makes sense to make the following changes:

* Do not force users to have to specify account_id, region_name, rest_api_id etc. I see this as a really low level configuration option and we should not be forcing constraints on the contents of the policy nor manipulating the contents.
* We should only support reading it from a file. While the `config.json` can get complex, I want to avoid it from being even more complex by adding support for arbitrary dictionaries as a value. This is also how we expose custom IAM policies for the lambda function.
* We rename the configuration option to `api_gateway_policy_file` to be consistent with `iam_policy_file`

Thoughts?

Reply:
the problem is a user does want to provide a scoped down policy, they need those values to be able to craft the arn correctly which is fundamental to providing custom iam policies, perhaps they have static hard codes for account id and region and do some additional devops to make it work across accounts/regions for principial arns etc, but its rather hard for them to supply rest_api_id outside of the framework providing it. its unclear if that is needed for scope down, most of the examples have it, but its embedded on the resource directly so it shouldn't really grant access to things other than itself.

re separate file sounds fine, but i'll note that lambda iam policy is fairly arbitrary in length and nature while this is fairly constrained from a usage & definition perspective.

overall the suggested changes from this round of reviews sounds good and should enable simplifying this a bit, thanks.

Reply:
>> unclear if that is needed for scope down, most of the examples have it, but its embedded on the resource directly so it shouldn't really grant access to things other than itself.

Yeah. If the APIG policy was a standalone resource that could be shared across APIG API's needing the rest_api_id, it would make sense. However given that the policy is not standalone and can only be attached to a single APIG API, it seems like the `account_id`, `region`, `api_id` is not really useful other than accidentally attaching the wrong policy to an API? In terms of scoping down access to the resources of an APIG API, you would really only want to do it for methods and routes: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html#api-gateway-iam-policy-resource-format-for-executing-api, in which case you can still use `*` for `region`, `account_id`, `api_id` and get the same result if you had specified the literal value instead.

Reply:
removed variable interpolation and switched to user specifying a policy file with the suggested configuration name.